src/vivarium_csu_hypertension_sdc/components/metrics.py

```diff
@@ -139,7 +140,7 @@ def setup(self, builder):
     def on_collect_metrics(self, event):
         # FIXME: is the timing right here? I always get confused about whether I'm looking at things at the right time
         pop = self.population_view.get(event.index).query('alive == "alive"')
-        pop = pop.loc[pop.last_prescription_date == event.time]
+        pop = pop.loc[(self.clock() < pop.last_prescription_date) & (pop.last_prescription_date <= event.time)]
```

Comment:
Do we have other similar checks that might be problematic?

Reply:
I don't think so. The only other place I do checks like this is about visit dates but those are always going to be actually on the time step so it shouldn't matter